# HG changeset patch
# User Andreas Weber <andy.weber.aw@gmail.com>
# Date 1449956639 -3600
#      Sat Dec 12 22:43:59 2015 +0100
# Node ID 7234cbd0429252473201e8668ead6c0a411de7c4
# Parent  40da964c3add342fe31801b3b651629bde90b43d
magic.m:Fix failing tests after cset cdf08aacfd86

diff --git a/scripts/special-matrix/magic.m b/scripts/special-matrix/magic.m
--- a/scripts/special-matrix/magic.m
+++ b/scripts/special-matrix/magic.m
@@ -30,16 +30,17 @@
 ## @end deftypefn
 
 function A = magic (n)
 
   if (nargin != 1)
     print_usage ();
   endif
 
+  n = fix (n);
   if (n < 1)
 
     A = [];
 
   elseif (mod (n, 2) == 1)
 
     shift = floor ((0:n*n-1)/n);
     c = mod ([1:n*n] - shift + (n-3)/2, n);
@@ -79,18 +80,16 @@ endfunction
 
 
 %!test
 %! for i = 3:30
 %!   A = magic (i);
 %!   assert (norm(diff([sum(diag(A)),sum(diag(flipud(A))),sum(A),sum(A')])),0);
 %! endfor
 
+%!assert (isempty (magic (-1)))
 %!assert (isempty (magic (0)))
 %!assert (magic (1), 1)
+%!assert (magic (1.5), 1)
 
 ## Test input validation
 %!error magic ()
 %!error magic (1, 2)
-%!error <N must be a positive integer not equal to 2> magic (1.5)
-%!error <N must be a positive integer not equal to 2> magic (-1)
-%!error <N must be a positive integer not equal to 2> magic (2)
-
